src/rpc/cli.cpp

```diff
@@ -198,10 +216,18 @@ void cli::getline( const fc::string& prompt, fc::string& line)
          line_read = readline(prompt.c_str());
          if( line_read == nullptr )
             FC_THROW_EXCEPTION( fc::eof_exception, "" );
-         if( *line_read )
-            add_history(line_read);
          line = line_read;
-         free(line_read);
+         try
+         {
+            if (*line_read)
+               add_history(line_read);
+            free(line_read);
```

Comment:
@pmconrad I guess it doesn't harm to try to `free` again when the first `free` threw an exception?

Reply:
It does. I'll fix it yet again.